types/json_test.go

```diff
@@ -37,7 +37,7 @@ func TestJSONBuilder(t *testing.T) {
 	require.Zero(t, b.Len(), "unexpected ArrayBuilder.Len(), NewJSONBuilder did not reset state")
 	require.Zero(t, b.Cap(), "unexpected ArrayBuilder.Cap(), NewJSONBuilder did not reset state")
 	require.Zero(t, b.NullN(), "unexpected ArrayBuilder.NullN(), NewJSONBuilder did not reset state")
-	require.Equal(t, `["{"a":1,"b":2}" (null) "{"c":3,"d":4}" (null) "{"e":5,"f":6}" "{"g":7,"h":8}"]`, a.String())
+	require.Equal(t, `["{\"a\":1,\"b\":2}" (null) "{\"c\":3,\"d\":4}" (null) "{\"e\":5,\"f\":6}" "{\"g\":7,\"h\":8}"]`, a.String())
```

Comment:
That doesn't seem right. `String()` should return a string representation for json.

Reply:
It isn't JSON, though, it is JSONArray.
And to be able to parse this properly we do need to use quoted strings.

Reply:
Specifically, see `ValueStr` test below